static/diagrams/membership.drawio

```diff
@@ -1 +1,159 @@
+        <mxCell id="P05zALBlSnZwpnZ5Cu9t-29" value="External &lt;br&gt;(not a member)" style="rounded=1;whiteSpace=wrap;html=1;fontFamily=Helvetica;" parent="1" vertex="1">
+          <mxGeometry x="50" y="430" width="120" height="60" as="geometry" />
+        </mxCell>
+        <mxCell id="P05zALBlSnZwpnZ5Cu9t-30" value="/knock if &lt;br&gt;knock-able&lt;br&gt;join_rules" style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;exitX=0.25;exitY=0;exitDx=0;exitDy=0;entryX=0.5;entryY=0;entryDx=0;entryDy=0;fontFamily=Helvetica;jumpStyle=none;jumpSize=7;" parent="1" source="P05zALBlSnZwpnZ5Cu9t-29" target="P05zALBlSnZwpnZ5Cu9t-24" edge="1">
+          <mxGeometry x="-0.7647" relative="1" as="geometry">
+            <Array as="points">
+              <mxPoint x="80" y="140" />
+              <mxPoint x="440" y="140" />
+            </Array>
+            <mxPoint as="offset" />
+          </mxGeometry>
+        </mxCell>
+        <mxCell id="P05zALBlSnZwpnZ5Cu9t-19" value="/ban" style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;exitX=0.25;exitY=0;exitDx=0;exitDy=0;entryX=0.5;entryY=0;entryDx=0;entryDy=0;jumpStyle=arc;" parent="1" source="P05zALBlSnZwpnZ5Cu9t-25" target="P05zALBlSnZwpnZ5Cu9t-3" edge="1">
+          <mxGeometry x="-0.9304" relative="1" as="geometry">
+            <Array as="points">
+              <mxPoint x="295" y="150" />
+              <mxPoint x="600" y="150" />
+            </Array>
+            <mxPoint as="offset" />
+          </mxGeometry>
+        </mxCell>
+        <mxCell id="P05zALBlSnZwpnZ5Cu9t-27" value="/leave&lt;br&gt;or /kick" style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;exitX=0.5;exitY=1;exitDx=0;exitDy=0;entryX=0.5;entryY=0;entryDx=0;entryDy=0;fontFamily=Helvetica;" parent="1" source="P05zALBlSnZwpnZ5Cu9t-24" target="P05zALBlSnZwpnZ5Cu9t-4" edge="1">
+          <mxGeometry x="-0.6" relative="1" as="geometry">
+            <mxPoint as="offset" />
+          </mxGeometry>
+        </mxCell>
+        <mxCell id="Je66lSOIGBRFVx_LtrYI-3" value="/knock if&lt;br&gt;knock-able&lt;br&gt;join_rules&lt;br&gt;" style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;exitX=0.75;exitY=0;exitDx=0;exitDy=0;entryX=0.75;entryY=1;entryDx=0;entryDy=0;" parent="1" source="P05zALBlSnZwpnZ5Cu9t-4" target="P05zALBlSnZwpnZ5Cu9t-24" edge="1">
+          <mxGeometry x="-0.4" y="-5" relative="1" as="geometry">
+            <mxPoint as="offset" />
+          </mxGeometry>
+        </mxCell>
+        <mxCell id="q5SATBPCpZFuv_SF1K4v-1" value="/join if join_rules&lt;br&gt;knock_restricted&lt;br&gt;and using a &lt;br&gt;restricted join" style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;exitX=0.25;exitY=0;exitDx=0;exitDy=0;entryX=0;entryY=0.25;entryDx=0;entryDy=0;jumpStyle=arc;" edge="1" parent="1" source="P05zALBlSnZwpnZ5Cu9t-24" target="P05zALBlSnZwpnZ5Cu9t-2">
```

Comment:
> /join if join_rules knock_restricted and using a restricted join

I assume by "restricted join", we mean "a join satisfying one of the conditions for [restricted rooms](https://spec.matrix.org/v1.9/client-server-api/#conditions)"?

The transition is also permitted if the join_rules are `restricted` (and the restriction conditions are satisfied), or as @Xiretza points out, if the join rules are `public`.

Indeed, I think the rules for the knock -> join transition are identical to those for the leave -> join and external -> join transitions (which makes sense intuitively, and makes sense because neither `leave` nor `knock` are specifically mentioned in auth rule 4.3, which covers the transition to `join` generally). So, can we just make it use the same label "/join if allowed by join_rules" ?